src/py/pyodide/console.py

```diff
@@ -479,19 +479,66 @@ async def runcode(self, source: str, code: CodeRunner) -> ConsoleFuture:
         return await super().runcode(source, code)
 
 
-def repr_shorten(
-    value: Any, limit: int = 1000, split: int | None = None, separator: str = "..."
+def shorten(
+    text: str, limit: int = 1000, split: int | None = None, separator: str = "..."
 ) -> str:
-    """Compute the string representation of ``value`` and shorten it
-    if necessary.
+    """Shorten ``text`` it if necessary.
 
     If it is longer than ``limit`` then return the firsts ``split``
     characters and the last ``split`` characters separated by '...'.
     Default value for ``split`` is `limit // 2`.
+    If ``split`` is longer than the length of ``len(value) // 2``, ``split`` will be `len(value) // 2`.
+    limit must be greater than or equal to 2. If it is less than 2, an error occurs.
+
+    Examples
+    --------
+    >>> from pyodide.console import shorten
+    >>> sep = "_"
+    >>> shorten("abcdefg", limit=5, separator=sep)
+    'ab_fg'
+    >>> shorten("abcdefg", limit=12, separator=sep)
+    'abcdefg'
     """
+    if limit < 2:
+        raise ValueError("limit must be greater than or equal to 2.")
     if split is None:
         split = limit // 2
-    text = repr(value)
+    split = min(split, len(text) // 2)
     if len(text) > limit:
         text = f"{text[:split]}{separator}{text[-split:]}"
     return text
+
+
+def repr_shorten(
+    value: Any, limit: int = 1000, split: int | None = None, separator: str = "..."
+) -> str:
+    """Compute the string representation of ``value`` and shorten it
+    if necessary.
+
+    If it is longer than ``limit`` then return the firsts ``split``
+    characters and the last ``split`` characters separated by '...'.
+    Default value for ``split`` is `limit // 2`.
+    If ``split`` is longer than the length of ``len(value) // 2``, ``split`` will be `len(value) // 2`.
+    limit must be greater than or equal to 4. If it is less than 4, an error occurs.
+
```

Comment:
```suggestion
	This is equivalent to `shorten(repr(value), limit, split, separator)`.
```

Reply:
looks so good! I've fixed them all. Thank you. 😊